### PR TITLE
Create Cycle_Sort.c

### LINKED ISSUE #2146: Cycle Sort
- [ ] C - @saloniankita
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @m-code12
- [x] Java
- [x] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by saloniankita**
Fixes #2146 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- C
- 

#### Files Added:
- Cycle_Sort.c
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Cycle_Sort.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#include<stdio.h>
```
Comment (MastersAbh) [inline Cycle_Sort.c:1]: Problem statement at top of file

File: Cycle_Sort.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                 write++;  
+            }  
+        }  
+    }  
+   
+ }  
+   
+int main()  
+{  
+    int a[] = {1, 9, 2, 4, 2, 10, 45, 3, 2};
```
Comment (MastersAbh) [inline Cycle_Sort.c:53]: Please take dynamic inputs

File: Cycle_Sort.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        if (pos == start)  
+            continue;  
+        
+        while (element == a[pos])  
+            pos += 1;  
+        
+        if (pos != start) 
+        {  
+        temp = element;
```
Comment (MastersAbh) [inline Cycle_Sort.c:24]: indentation

File: Cycle_Sort.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    int n = sizeof(a) / sizeof(a[0]), i;  
+    sort(a, n);  
+    printf(" After sorting : ");  
+    for (i = 0 ; i < n ; i++)  
+        printf("%d  ",a[i]);  
+    
+}  
+
+
+
```
Comment (MastersAbh) [inline Cycle_Sort.c:63]: only one blank line is enough

**Comment by saloniankita** [comment]
Done @MastersAbh

File: Cycle_Sort.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
+#include <stdio.h>
```
Comment (atarax665) [inline Cycle_Sort.c:1]: Please do add a synopsis of cycle sort in the beginning.

**Comment by saloniankita** [comment]
Done @atarax665

**Comment by saloniankita** [comment]
Kindly review this PR @MastersAbh @atarax665

File: Cycle_Sort.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            if (element != a[pos]) 
+            {  
+                 temp = element;  
+                 element = a[pos];  
+                 a[pos] = temp;    
+                 write++;  
+            }  
+        }  
+    }  
+
```
Comment (atarax665) [inline Cycle_Sort.c:56]: Remove this line

**Comment by saloniankita** [comment]
Done @atarax665

**Pull request merged**